beaver/transports/stomp_transport.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+import stomp
+
+from beaver.transports.base_transport import BaseTransport
+from beaver.transports.exception import TransportException
+
+
+class StompTransport(BaseTransport):
+
+    def __init__(self, beaver_config, logger=None):
+        """
+        Mosquitto client initilization. Once this this transport is initialized
+        it has invoked a connection to the server
+        """
+        super(StompTransport, self).__init__(beaver_config, logger=logger)
+
+        self.createConnection(beaver_config)
+        self.logger = logger
+
+ 
+    def callback(self, filename, lines, **kwargs):
+        """publishes lines one by one to the given topic"""
+        timestamp = self.get_timestamp(**kwargs)
+        if kwargs.get('timestamp', False):
+            del kwargs['timestamp']
+        
+
+        for line in lines:
+            try:
+                import warnings
+                with warnings.catch_warnings():
+                    warnings.simplefilter('error')
+                    m = self.format(filename, line, timestamp, **kwargs)
+                    self.logger.debug("Sending message " + m)
+                    self.conn.send(destination=self.queue, body=m)    
```

Comment:
Ah,

I see that. Stomp does not permit this so I cannot do this change.

Putting transactions around the batch of lines is only more efficient if there are more than 3 lines in a batch and then it's a bit debatable wether it would be quicker as each begin and commit requires a network round trip.

However have you thought about adding a Json_lines option (like logstash) which would batch all the lines into 1 message making it efficient for all transports (not just stomp) ? I can look at introducing this if you want but I think it would change the definition of the BaseTransport class.

Peter